1. Basic Web Development/frontend/src/main.ts

```diff
@@ -0,0 +1,419 @@
+
+            createPermissionsForm(form);
+
+            fetch(API_URL + '/pref/delete', { method: 'DELETE', headers: HEADERS, body: JSON.stringify({ name: target.name }) })
+                .then(response => response.json())
+                .catch((error) => {
+                    console.error('Error:', error);
+                    permissionsRoutine();
+                });
+        }
+    });
+}
+
+function createPermissionsForm(form: HTMLElement) {
+    const permissionsList = document.getElementById("permissions-list");
+
+    removeChildren(permissionsList);
+
+    this.PERMISSIONS.forEach((element: Permission) => {
+        const newElement = document.createElement("div");
+        newElement.classList.add("grid-element");
+
+        const newLabel = document.createElement("label");
+        newLabel.setAttribute('for', element.id);
+        newLabel.innerHTML = element.text;
+
+        const newInput = document.createElement("input");
+        newInput.setAttribute('id', element.id);
+        newInput.setAttribute('name', element.type);
+        newInput.setAttribute('type', 'checkbox');
+
+        if (element.type == "permissionNss") {
+            newInput.click();
+            newInput.disabled = true;
+        }
+
+        const newButtonDelete = document.createElement("button");
+        newButtonDelete.setAttribute('type', 'button');
+        newButtonDelete.textContent = 'X';
+        newButtonDelete.setAttribute('name', element.id);
+        newButtonDelete.classList.add('letter-btn');
+        newButtonDelete.classList.add('delete-btn');
+
+        const newButtonEdit = document.createElement("button");
+        newButtonEdit.setAttribute('type', 'button');
+        newButtonEdit.textContent = 'E';
+        newButtonEdit.setAttribute('name', element.id);
+        newButtonEdit.classList.add('letter-btn');
+        newButtonEdit.classList.add('edit-btn');
+
+        newElement.appendChild(newButtonEdit);
+        newElement.appendChild(newButtonDelete);
+        newElement.appendChild(newLabel);
+        newElement.appendChild(newInput);
+
+        permissionsList.appendChild(newElement);
+    });
+
+    setCounter();
+
+    this.BOXES = getBoxes();
+}
+
+function setCounter() {
+    const counterDisplay = document.getElementById("counter-display");
+    removeChildren(counterDisplay);
+
+    this.COUNTER = document.querySelectorAll('input[type="checkbox"]:checked').length;
+
+    const newText = document.createElement('span');
+    newText.textContent = `${this.COUNTER} permissions`;
+    counterDisplay.appendChild(newText);
+}
+
+function handleRadios() {
+    document.addEventListener('change', event => {
+        if ((event.target as HTMLInputElement).name == "preset") {
+            switch ((event.target as HTMLInputElement).value) {
+                case "All":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, true);
```

Comment:
Calling `this.changeBoxesState` seems to throw a console error. Any idea as to why that happens?

Reply:
I had no idea why it was happening but It's fixed on the new branch

Reply:
ok, can you still try and figure it out? We'd like to see that you understand why.